tests/test_integration.py

```diff
@@ -872,6 +901,295 @@ async def test_send_message_return_timestamp(src: str, dest: str, swarm7: dict[s
+            rejected_value_before = balance_str_to_int(statistics_before.rejected_value)
+
+            # the destination should receive all the packets
+            packets = [f"Standard ticket win probability check: #{i:08d}" for i in range(ticket_count)]
+            await send_and_receive_packets_with_pop(
+                packets, src=swarm7[src], dest=swarm7[dest], path=[swarm7[relay].peer_id]
+            )
+
+            # in this case, the relay has tickets for all the packets, because the source sends them with win prob = 1
+            ticket_statistics = await swarm7[relay].api.get_tickets_statistics()
+            unredeemed_value = balance_str_to_int(ticket_statistics.unredeemed_value)
+            rejected_value = balance_str_to_int(ticket_statistics.rejected_value)
+            assert unredeemed_value - unredeemed_value_before == TICKET_PRICE_PER_HOP * ticket_count
+            assert ticket_statistics.winning_count - statistics_before.winning_count == ticket_count
+            assert rejected_value - rejected_value_before == 0
+
+            # at this point the tickets become neglected, since the channel will be closed
+    finally:
+        # Always return winning probability to 1.0 even if the test failed
+        set_minimum_winning_probability_in_network(private_key, 1.0)
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "route",
+    [
+        [
+            *random.sample(nodes_with_lower_outgoing_win_prob(), 1),
+            *random.sample(barebone_nodes(), 2),
+        ]
+        for _ in range(PARAMETERIZED_SAMPLE_SIZE)
+    ],
+)
+async def test_hoprd_should_not_accept_tickets_with_lower_than_min_win_prob(route, swarm7: dict[str, Node]):
+    ticket_count = 10
+
+    src = route[0]
+    relay = route[1]
+    dest = route[-1]
+
+    async with create_channel(swarm7[src], swarm7[relay], funding=ticket_count * TICKET_PRICE_PER_HOP):
+        # ensure ticket stats are what we expect before starting
+        statistics_before = await swarm7[relay].api.get_tickets_statistics()
+        unredeemed_value_before = balance_str_to_int(statistics_before.unredeemed_value)
+
+        # sent out all packets at from source
+        packets = [f"Rejected ticket win probability check: #{i:08d}" for i in range(ticket_count)]
+        random_tag = gen_random_tag()
+
+        for packet in packets:
+            assert await swarm7[src].api.send_message(swarm7[dest].peer_id, packet, [swarm7[relay].peer_id], random_tag)
+
+        # in this case, the destination receives nothing, because the relayer will not relay packets
+        # with win prob lower than 1
+        # TODO: how long do we need to wait before we can safely conclude that destination has received nothing?
+        await asyncio.sleep(5.0)
+        response = await swarm7[dest].api.messages_peek_all(random_tag)
+        assert response is not None
+        assert len(response.messages) == 0
```

Comment:
How does the suggested code solve the problem?

The issue here is, that one cannot be sure how long to wait before it can be safely concluded that no messages were received.

Reply:
This approach only addresses a minor problem of failing fast.
However, it does not solve the major concern - that is the fact that we don't know what timeout period would be reasonable to conclude that no messages were received.

Reply:
Good point